include/onnxruntime/core/graph/rewrite_rule.h

```diff
@@ -54,6 +69,63 @@ class GraphEditor {
     return graph_.Resolve();
   }
 
+  // TODO: The following functions should probably be moved to the Graph and Node classes.
+  // We also need to decide whether we keep the GraphEditor or we pass directly the Graph.
+  // If we keep it, it might be better to put it in a separate file.
+
+  /** Check whether the node has a single input and a single output. */
+  bool IsSingleInSingleOutNode(const Node& node) const {
+    return node.GetInputEdgesCount() == 1 && node.GetOutputEdgesCount() == 1;
+  }
+
+  /** Return the attribute of a Node. */
+  const onnx::AttributeProto* GetNodeAttribute(
```

Comment:
Same as above.

These looks to me are "helper" functions. They should not be taken as instance methods of GraphEditor, Graph or Node.

Reply:
Similar to the function above, I think these are useful functions to have (they will be useful for any transformation). I agree GraphEditor is not the best place though. The Graph class is a better fit.

Reply:
no. please don't do that. A way to check whether it's an instance method of a class is whether you're using (*this) in the method implementation.

Reply:
Clarification: I meant adding it to the Node class, not the Graph class. All these methods do get the Node as an argument in their implementation. 
So, yes, if we kept them in the GraphEditor, they should have been static. But it is better to move them to the Node class, given that they are useful in general.